src/components/legend/get_legend_data.js

```diff
@@ -66,31 +69,41 @@ module.exports = function getLegendData(calcdata, opts) {
     // won't draw a legend in this case
     if(!lgroups.length) return [];
 
+    // collapse all groups into one if all groups are blank
+    var shouldCollapse = !hasOneNonBlankGroup || !grouped;
+
     // rearrange lgroupToTraces into a d3-friendly array of arrays
-    var lgroupsLength = lgroups.length;
-    var ltraces;
     var legendData;
 
-    if(hasOneNonBlankGroup && helpers.isGrouped(opts)) {
-        legendData = new Array(lgroupsLength);
-
-        for(i = 0; i < lgroupsLength; i++) {
-            ltraces = lgroupToTraces[lgroups[i]];
-            legendData[i] = helpers.isReversed(opts) ? ltraces.reverse() : ltraces;
+    legendData = [];
+    for(i = 0; i < lgroups.length; i++) {
+        var t = lgroupToTraces[lgroups[i]];
+        if(shouldCollapse) {
+            legendData.push(t[0]);
+        } else {
+            legendData.push(t);
         }
-    } else {
-        // collapse all groups into one if all groups are blank
-        legendData = [new Array(lgroupsLength)];
-
-        for(i = 0; i < lgroupsLength; i++) {
-            ltraces = lgroupToTraces[lgroups[i]][0];
-            legendData[0][helpers.isReversed(opts) ? lgroupsLength - i - 1 : i] = ltraces;
+    }
+    if(shouldCollapse) legendData = [legendData];
+
+    var orderFn = function(a, b) {
+        return a.trace.legendrank - b.trace.legendrank;
+    };
+    for(i = 0; i < legendData.length; i++) {
```

Comment:
```suggestion
    var orderFn = function(a, b) {
        return (a.trace.legendrank - b.trace.legendrank) || (a._index - b._index);
    };
    for(i = 0; i < legendData.length; i++) {
        legendData[i].forEach(function(a, i) { a._index = i; });
```

Does this fix the old-chrome stability issue?

Reply:
Thanks. Applied in https://github.com/plotly/plotly.js/pull/5591/commits/d35ff08123ad49388ec565fba174a83f9f935563.